src/EditorFeatures/Test2/ReferenceHighlighting/CSharpReferenceHighlightingTests.vb

```diff
@@ -945,5 +945,262 @@ namespace X
 
             Await VerifyHighlightsAsync(input, testHost)
         End Function
+
+        <WpfTheory, CombinatorialData>
+        <WorkItem(42988, "https://github.com/dotnet/roslyn/issues/42988")>
+        Public Async Function TestNintNUint1(testHost As TestHost) As Task
+            Dim input =
+            <Workspace>
+                <Project Language="C#" CommonReferences="true">
+                    <Document>
+using System;
+class Test
+{
+    void M()
+    {
+        $${|Reference:nint|} n1 = 1;
```

Comment:
Why the behavior changes when `CommonReferencesNet7=true`?

Reply:
Oh I just realize this is the design decision of .net 7

Reply:
:)

Reply:
How does this actually work when the runtime is different? (i.e. I am surprised with how this work, are the symbols we get from the compiler side different?)

Reply:
So the compiler produces different symbols.  They differ with this property:

https://github.com/dotnet/roslyn/blob/main/src/Compilers/Core/Portable/Symbols/ITypeSymbol.cs#L82

On .net7 and above, both of these will return true for IsNativeIntegerType.  Prior to .net 7, only nint/nuint would, and IntPtr/UIntPtr would return false.